Mu2eUtilities/src/TriggerResultsNavigator.cc

```diff
@@ -41,11 +41,26 @@ namespace mu2e {
     }  
 
     //loop over trigResults to fill the map <string, unsigned int)
+    std::string   delimeter=":";
     for (unsigned int i=0; i< _trigPathsNames.size(); ++i){
-      _trigMap.insert(std::pair<std::string, unsigned int>(_trigPathsNames[i], i));
+      size_t       pos      = _trigPathsNames[i].find(delimeter);
+      unsigned int bit      = std::stoi(_trigPathsNames[i].substr(0, pos));
+      std::string  pathName = _trigPathsNames[i].substr(pos+1, _trigPathsNames[i].length());
+      _trigMap.insert(std::pair<std::string, unsigned int>(pathName, i));
+      _trigPathMap.insert(std::pair<std::string, unsigned int>(pathName, bit));
     }
   }
 
+
+  std::string const
+  TriggerResultsNavigator::getTrigPathName(unsigned int const i) const
+  {
+    std::string   delimeter =":";
+    size_t        pos       = _trigPathsNames[i].find(delimeter);
+    if (pos > _trigPathsNames[i].length()) return "TRIG PATH NOT FOUND";
+    return _trigPathsNames[i].substr(pos+1, _trigPathsNames[i].length());
```

Comment:
How will this function be used?  I generally don't like returning a magic string to denote an error.  But there are probably use cases in which it is the best way - for example if it's only being used in printout and it's normal that it might sometimes be missing.

Reply:
I honestly wasn't happy either, but I thought that there might be analysis code that could be ok with having a pathname not found.

Reply:
The standard answer is to define a small handle-like class and return that.  It would have two data members, the string and bool isValid.  The data should be private and it should throw if you someone tries to access an invalid string.

I have asked Marc and Kyle to see if a standard one exists.  If not I can write one as a template and you can use it.  No need to hold up this PR.